### PR TITLE
Feature/discord guild

### PR DESCRIPTION
**Pull request by fisher60**

### CODE REVIEW

File: friendo_site/friendo_site/graphql/schemas/guilds.graphql [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+input GuildInput {
+    guild_id: BigInt!
+}
+
+input GuildUpdateInput {
+    guild_id: BigInt!
+    dogeboard_id: BigInt
+    dogeboard_emoji: String
+    dogeboard_reactions_required: Int
+}
```
Comment (GDWR) [inline friendo_site/friendo_site/graphql/schemas/guilds.graphql:17]: Would this allow for passing initial data into the creation of the Guild?

File: friendo_site/friendo_site/graphql/types/guilds.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+def get_guild(_, info, data):
+    if data.get("guild_id", None):
+        return Guild.objects.get(guild_id=data.get("guild_id", None))
+    else:
+        raise KeyError("guild_id is missing")
+
+
+@token_required
+def create_guild(_, info, data):
+    if data.get("guild_id", None):
```
Comment (GDWR) [inline friendo_site/friendo_site/graphql/types/guilds.py:15]: Walrus operator 👀

File: friendo_site/friendo_site/graphql/types/guilds.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+def update_guild(_, info, data):
+    if data.get("guild_id", None):
+        this_guild = Guild.objects.get(guild_id=data.get("guild_id", None))
+        if data.get("dogeboard_id", None):
+            this_guild.dogeboard_id = int(data.get("dogeboard_id", None))
+        if data.get("dogeboard_emoji", None):
+            this_guild.dogeboard_emoji = data.get("dogeboard_emoji", None)
+        if data.get("dogeboard_reactions_required", None):
+            this_guild.dogeboard_reaction_required = int(
+                data.get("dogeboard_reactions_required", None)
```
Comment (GDWR) [inline friendo_site/friendo_site/graphql/types/guilds.py:33]: Walrus Operator?

File: friendo_site/friendo_site/guilds/models.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+from django.db import models
+
+
+class Guild(models.Model):
+    guild_id = models.BigIntegerField(unique=True)
+    dogeboard_id = models.BigIntegerField(blank=True, null=True)
+    dogeboard_emoji = models.CharField(max_length=60, blank=True, null=True)
```
Comment (GDWR) [inline friendo_site/friendo_site/guilds/models.py:7]: can we default to 🐶 for the dogeboard_emoji? 
I am doing this on the discord cog but might be handy.

File: friendo_site/friendo_site/graphql/types/guilds.py (hunk 3 above)
Comment (GDWR) [reply friendo_site/friendo_site/graphql/types/guilds.py:33]: `
    if guild_id := data.get("guild_id", None):
        this_guild = Guild.objects.get(guild_id=guild_id)
        if dogeboard_id := data.get("dogeboard_id", None):
            this_guild.dogeboard_id = int(dogeboard_id)
        if dogeboard_emoji := data.get("dogeboard_emoji", None):
            this_guild.dogeboard_emoji = dogeboard_emoji 
        if dogeboard_reactions_required := data.get("dogeboard_reactions_required", None):
            this_guild.dogeboard_reaction_required = int(dogeboard_reactions_required )
`

Comment (GDWR) [reply friendo_site/friendo_site/graphql/types/guilds.py:33]: Although I would question the need for `int()` conversions

File: friendo_site/friendo_site/graphql/types/guilds.py (hunk 2 above)
Comment (GDWR) [reply friendo_site/friendo_site/graphql/types/guilds.py:15]:
```py
if guild_id := data.get("guild_id", None):
```

File: friendo_site/friendo_site/guilds/models.py (hunk 4 above)
Comment (fisher60) [reply friendo_site/friendo_site/guilds/models.py:8]: I am not 100% sure how the api will handle text input of emojis versus how we will save them with d.py. So I might just ignore this for now 🤷

**Pull request merged**